### ISSUE TITLE
"conf", "logits", and "probs" are all NaN

### ISSUE THREAD

**Issue by chrisole** [adamxyang/laplace-lora#1]
Hi! Thanks for your work.
I try to reproduce 'run_gpt.py' on several datasets. 
However, the 'eval_res.json' shows: {"index": 0, "true": 0, "pred": 0, "conf": NaN, "logits": [NaN, NaN], "probs": [NaN, NaN]} from the first time save the checkpointss. I try different learning rates but the results are the same.
Foundation model is llama2-7b-chat. 
Do you know the reason?  Thanks!

**Comment by adamxyang** [comment]
Hi, thanks for your interests! It looks like a NaN error occurs during training, could you check if the model's outputs and loss go NaN during training? I have previously experienced similar issues when training reward models with PEFT, there seems to be some issue with [PEFT version and weights' precision](https://github.com/huggingface/peft/issues/1249#issuecomment-1853388308) that can cause NaN during training, you could also try adding `torch_dtype=torch.bfloat16` when loading the model.

**Comment by chrisole** [comment]
Many thanks for your reply! 
By the way, may I ask if this method also adaptive to auto-regressive languge generation? I noticed that these reasoning datasets are all like classification. I plan to extent your approach to longer sequence generation (e.g., Q&A task), is it feasible？I would appreciate if you give some advice : )

**Comment by adamxyang** [comment]
The current linearised Laplace framework only looks at uncertainty on a single prediction token, this may not translate easily to sentence-level uncertainty, and computing Jacobians could be quite costly. Currently, we're only working on extending the single token uncertainty to other settings, e.g. [reward modelling](https://arxiv.org/abs/2402.13210). Also we have released a new more efficient and self-contained library [here](https://github.com/MaximeRobeyns/bayesian_lora), that we plan to maintain in the future.

**Comment by chrisole** [comment]
Thanks a lot!
